test/jasmine/tests/carpet_test.js

```diff
@@ -0,0 +1,432 @@
+// var Plotly = require('@lib/index');
+var Plots = require('@src/plots/plots');
+// var Lib = require('@src/lib');
+
+var Carpet = require('@src/traces/carpet');
+var smoothFill2D = require('@src/traces/carpet/smooth_fill_2d_array');
+var smoothFill = require('@src/traces/carpet/smooth_fill_array');
+// var calc = require('@src/traces/carpet/calc');
+
+var customMatchers = require('../assets/custom_matchers');
+
+// var d3 = require('d3');
+// var createGraphDiv = require('../assets/create_graph_div');
+// var destroyGraphDiv = require('../assets/destroy_graph_div');
+// var customMatchers = require('../assets/custom_matchers');
+
+var test = {
+    supplyDefaults: false,
+    smoothFill2D: true
+};
+
+describe('carpet supplyDefaults', function() {
+    if(!test.supplyDefaults) return;
+    'use strict';
+
+    var traceIn,
+        traceOut;
+
+    var supplyDefaults = Carpet.supplyDefaults;
+
+    var defaultColor = '#444',
+        layout = {
+            font: Plots.layoutAttributes.font
+        };
+
+    beforeEach(function() {
+        traceOut = {};
+    });
+
+    it('uses a, b, x, and y', function() {
+        traceIn = {
+            a: [0, 1],
+            b: [0, 1, 2],
+            x: [[1, 2, 3], [4, 5, 6]],
+            y: [[2, 3, 4], [5, 6, 7]]
+        };
+
+        supplyDefaults(traceIn, traceOut, defaultColor, layout);
+
+        expect(traceOut.a).toEqual([0, 1]);
+        expect(traceOut.b).toEqual([0, 1, 2]);
+        expect(traceOut.x).toEqual([[1, 2, 3], [4, 5, 6]]);
+        expect(traceOut.y).toEqual([[2, 3, 4], [5, 6, 7]]);
+        expect(traceOut.da).toBeUndefined();
+        expect(traceOut.db).toBeUndefined();
+        expect(traceOut.a0).toBeUndefined();
+        expect(traceOut.b0).toBeUndefined();
+        expect(traceOut.visible).toBe(true);
```

Comment:
If you're already editing it, then 👍 . I think most of the carpet tests should actually be moved out of this file since some of the functions are better suited for lib.